### ISSUE TITLE
Two problems in the context of accessibility requirements in Germany

### ISSUE THREAD

**Issue by ATAschert** [keepassxreboot/keepassxc#8265]
## Overview
If you want to introduce KeePassXC in the state of Berlin, requirements regarding accessibility must be met according to the legal basis. The minimum requirements contain two points that cannot be met according to my tests. Missing alternative texts for menu icons and keyboard traps in the menu Create new passwords just interfere with the introduction of KeePassXC.

Thanks a lot | Vielen Dank

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
### Keyboard trap:
1. Open the dialog for creating a new password.
2. All steps must be fully accessible with the keyboard according to the regulation.
3. Neither Ctrl + S nor the Tab key can be used to reach the OK button for saving the password. The Tab key in the Notes field only adds tabs and does not jump to the next control.
![KeePassXC1](https://user-images.githubusercontent.com/88592890/178696305-6805f215-f3ff-4571-a736-05759503f19c.png)

### Alternative texts for menu icons
1. Activate the Narrator function in the Windows system settings.
2. When moving the mouse, no alternative text is read out for the icons in the menu bar (Open, Save, Add, ...), so that visually impaired people cannot orientate themselves by these.
![KeePassXC2](https://user-images.githubusercontent.com/88592890/178696339-417b5b7a-4b01-477a-a036-e7a5898786cd.png)

## Context
[NOTE]: # ( Give us any additional information you may have. )
Requirements based on EN 301 549.

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.7.1
Revision: 5916a8f

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows Server 2016 Version 1809

**Comment by vimpostor** [comment]
> 1. Open the dialog for creating a new password.
> 2. All steps must be fully accessible with the keyboard according to the regulation.
> 3. Neither Ctrl + S nor the Tab key can be used to reach the OK button for saving the password. The Tab key in the Notes field only adds tabs and does not jump to the next control.

Hitting Ctrl + Enter will trigger the OK button. I fear that some users probably want to use the Tab key inside the notes, but we could implement triggereing OK with Ctrl + S too, I guess.

**Comment by droidmonkey** [comment]
The notes field failure should be filed as a bug against qt, we just use the standard QTextEdit and allow tabs because people expect that feature. They don't offer a way to escape via keyboard that works across all platforms.

Do note you can press CTRL+TAB to navigate backwards out of the Notes field and wrap around to the OK button. And there are alternative shortcuts to leave the dialog itself:

CTRL+ENTER = OK
ESCAPE = CANCEL

**Comment by ATAschert** [comment]
Thank you very much! That helps a lot already. As for the other problem, I'm now helping myself to a transitional solution, so that all functions are also accessible via the menus and the narrator works here without any problems. Then the software can already be introduced. Nevertheless, I would be happy if this would be supplemented in a next version of KeePassXC with alternative text.

Once again, many thanks for the quick help. Greetings from Berlin.

**Comment by droidmonkey** [comment]
Good news, we will keep this open to track the issues though.

**Comment by newhinton** [comment]
@ATAschert I don't know if this is relevant, but i found out that the passphrase-generator also provides english only. Maybe keepass should also try to include other dictionaries for other languages. 

Should that be a new issue?

**Comment by droidmonkey** [comment]
We allow you to load your own passphrase dictionary in 2.7.0+, so no.

**Comment by DVZteama11y** [comment]
Hi everyone,
we are evaluating some software for our company. And KeePassXC is one of our favorites. But we need some statements about the accesibility status of KeePassXC. ATAschert mentioned the EN 301549 in an earlier post. The EN 301549 is very important in Germany. We have to comply these standards. So, we would like to ask you, if you have an accessibilty test or a [VPAT ](https://en.wikipedia.org/wiki/Voluntary_Product_Accessibility_Template) of KeePassXC? In general, how do you anchor accessibility in the software?

We would be very pleased to receive your answers, opinions and concrete proposals to the topic asseccibility.
If you need more information so please ask. :)
Greetings from Germany

**Comment by droidmonkey** [comment]
All of our accessibility is based on Qt 5, so you can start there to get a baseline for what is available by default. We had an AAA assessment conducted many years ago that allowed us to close numerous accessibility problems. The majority of which were colors and narration. At this time, besides the toolbar narration above, we are not aware of any accessibility problems with the software. We do not have compliance documentation, so you'll have to do that yourself for now. 

We will gladly address any accessibility issues that are found to be out of compliance with standards.

**Comment by droidmonkey** [comment]
I looked into the second issue, narration of the toolbar items, and there is nothing we can do about that. All toolbar items are represented in the main menu drop downs which do support narration; therefor this is not a breaking finding for accessibility. Qt does not support narration of the toolbar. Your first issue is tracked by #9060 and will be fixed in an upcoming PR.
